server/audit_math/suite.py

```diff
@@ -1,718 +0,0 @@
-                    * (cumulative_sample[loser] / prev_sample_size)
-                ),
-                bp_stratum.vote_totals[loser],
-            ),
-        }
-
-    hyp_sample_size = n2
-    hyp_sample_size = min(bp_stratum.num_ballots, hyp_sample_size)
-
-    for rnd in bp_stratum.sample:
-        hyp_sample[rnd] = bp_stratum.sample[rnd]
-
-    hyp_no_cvr_stratum = BallotPollingStratum(
-        bp_stratum.num_ballots, bp_stratum.vote_totals, hyp_sample, hyp_sample_size
-    )
-
-    return maximize_fisher_combined_pvalue(
-        alpha, contest, hyp_no_cvr_stratum, hyp_cvr_stratum, winner, loser
-    )
-
-
-def get_sample_size(
-    risk_limit: int,
-    contest: Contest,
-    bp_stratum: BallotPollingStratum,
-    cvr_stratum: BallotComparisonStratum,
-) -> HybridPair:
-    """
-    Estimate the initial sample sizes for the audit.
-
-    Inputs:
-        risk_limit      - the risk limit (as an integer percentage) for this audit
-        contest         - the overall contest information
-        bp_stratum: a stratum object containing the ballot polling stratum
-        cvr_stratum: a stratum object containing the ballot comparison stratum
-    Outputs:
-        sample_sizes    - A Tuple of (cvr_strata_size, no_cvr_strata_size).
-    """
-
-    alpha = float(risk_limit) / 100
-
-    for winner, loser in product(contest.winners, contest.losers):
-        n_ratio = cvr_stratum.num_ballots / (
-            cvr_stratum.num_ballots + bp_stratum.num_ballots
-        )
-        num_sampled = max(
-            MIN_SAMPLE_SIZE, cvr_stratum.sample_size + bp_stratum.sample_size
-        )
-
-        ballots_to_sample = num_sampled
-
-        expected_pvalue = 1.0
-
-        # this allows us to exactly match CORLA18's estimate_n and estimate_escalation_n
-        coefficient = 1.1
-        if bp_stratum.sample_size == 0 and cvr_stratum.sample_size == 0:
-            coefficient = 2.0
-
-        # step 1: linear search, increasing n by a factor of 1.1 each time
-        while (expected_pvalue > alpha) or (expected_pvalue is np.nan):
-            ballots_to_sample = int(coefficient * ballots_to_sample)
-            if ballots_to_sample > contest.ballots:
-                cvr_ballots_to_sample = math.ceil(n_ratio * contest.ballots)
-                bp_ballots_to_sample = int(contest.ballots - cvr_ballots_to_sample)
-                return HybridPair(
-                    cvr=cvr_ballots_to_sample, non_cvr=bp_ballots_to_sample
-                )
-
-            expected_pvalue = try_n(
-                ballots_to_sample,
-                alpha,
-                contest,
-                winner,
-                loser,
-                bp_stratum,
-                cvr_stratum,
-                n_ratio,
-            )
-
-        # step 2: bisection between n/1.1 and n
-        low_n = ballots_to_sample / coefficient
-        high_n = ballots_to_sample
-        mid_pvalue = 1.0
-        # TODO: do we need this tolerance?
```

Comment:
i think i understand the intention behind it - to provide a lower bound for the expected p value so that we don't end up with a bigger sample than necessary to meet the risk limit

Reply:
yeah, that does seem like the tradeoff. i wonder if there's a way to simulate with/without and get more info about it's usefulness